src/runtime/providers/math.ts

```diff
@@ -55,64 +94,153 @@ class Divide extends TotalFunctionConstraint {
-  static AttributeMapping = {
-    "degrees": 0,
-    "radians": 1,
-    "angle": 2
+class Sin extends TrigConstraint {
+  getReturnValue(args) {
+      return Math.sin(this.resolveTrigAttributes(args));
   }
+}
 
+class Cos extends TrigConstraint {
   getReturnValue(args) {
-    let [degrees, radians, angle] = args;
-    if (angle !== undefined) {
-      return this.getAngle(angle);
-    } else if (degrees !== undefined) {
-      return Math.sin(degrees * (Math.PI / 180));
-    } else {
-      return Math.sin(radians);
-    }
+      return Math.cos(this.resolveTrigAttributes(args));
   }
+}
 
-  @deprecated('Please use degrees instead of angle')
-  getAngle(angle) {
-    return Math.sin(angle * (Math.PI / 180));
+class Tan extends TrigConstraint {
+  getReturnValue(args) {
+      return Math.tan(this.resolveTrigAttributes(args));
+  }
+}
+
+class ASin extends ValueOnlyConstraint {
+  getReturnValue(args) {
+    return Math.asin(args[0]);
+  }
+}
+
+class ACos extends ValueOnlyConstraint {
+  getReturnValue(args) {
+    return Math.acos(args[0]);
+  }
+}
+
+class ATan extends ValueOnlyConstraint {
+  getReturnValue(args) {
+    return Math.atan(args[0]);
   }
 }
 
-class Cos extends TotalFunctionConstraint {
+
+class ATan2 extends TotalFunctionConstraint {
   static AttributeMapping = {
-    "degrees": 0,
-    "radians": 1,
-    "angle": 2
+    "x": 0,
+    "y": 1
+  }
+  getReturnValue(args) {
+    return [Math.atan2(args[0] ,args[1])];
   }
+}
 
+//Hyperbolic Functions
+class SinH extends ValueOnlyConstraint {
+  sinh (x: number):number{
+    var y = Math.exp(x);
+    return (y - 1 / y) / 2;
+  }
+  getReturnValue(args) {
+    return [this.sinh(args[0])];
+  }
+}
+
+class CosH extends ValueOnlyConstraint {
+  cosh (x: number):number{
+    var y = Math.exp(x);
+    return (y + 1 / y) / 2;
+  }
+  getReturnValue(args) {
+    return [this.cosh(args[0])];
+  }
+}
+
+class TanH extends ValueOnlyConstraint {
+  tanh(x : number) : number {
+    if (x === Infinity) {
+      return 1;
+    } else if (x === -Infinity) {
+      return -1;
+    } else {
+      let y = Math.exp(2 * x);
+      return (y - 1) / (y + 1);
+    }
+  }
   getReturnValue(args) {
-    let [degrees, radians, angle] = args;
-    if (angle !== undefined) {
-      return this.getAngle(angle);
-    } else if (degrees !== undefined) {
-      return Math.cos(degrees * (Math.PI / 180));
+    return [this.tanh(args[0])];
+  }
+}
+
+//Inverse Hyperbolic
+class ASinH extends ValueOnlyConstraint {
+  asinh (x: number):number{
+    if (x === -Infinity) {
+      return x;
     } else {
-      return Math.cos(radians);
+      return Math.log(x + Math.sqrt(x * x + 1));
     }
   }
+  getReturnValue(args) {
+    return [this.asinh(args[0])];
+  }
+}
 
-  @deprecated('Please use degrees instead of angle')
-  getAngle(angle) {
-    return Math.cos(angle * (Math.PI / 180));
+class ACosH extends ValueOnlyConstraint {
+  acosh (x: number):number{
+    //How do we handle number outside of range in Eve? 
```

Comment:
No, more a feed back issue. Currently I'm returning NaN, but the language has as yet no formal type system, so in Eve programs NaN or Infinity come out as strings, which can be tested for. If there's ever a type system this can be cleaned up, but ACosH and ATanH functions only work for a certain range of numbers.